### DISCUSSION TITLE
Does it make sense to add more detailed docs for some topics?

### DISCUSSION THREAD

**Discussion by ArthurHarkivsky** [micronaut-projects/micronaut-data#2600] [Q&A] [answered]
Hello!
Does it make sense to add more detailed docs for some topics?

For example, [https://guides.micronaut.io/latest/micronaut-data-jdbc-repository-gradle-java.html] or other topic (I am not sure)(https://guides.micronaut.io/latest/micronaut-data-jdbc-repository-gradle-java.html) should have info about:
- joins using Micronaut JDBC lib
- native queries with pagination using Micronaut JDBC lib
- Criteria API alternative in Micronaut JDBC lib
Or I missed it?
More docs -> less time spent in Micronaut team on support.

If some info is absent on different topics also, community will grow slowly, because a possible Micronaut user will not use it, if he is not see all required info in docs.
**Micronaut is great, it should grow faster!**

**Comment by radovanradic** [comment]
Maybe you missed Micronaut Data guide: https://micronaut-projects.github.io/micronaut-data/latest/guide/ ? It has info about joins, criteria, native queries etc.

**Reply by ArthurHarkivsky** [reply]
no, it is missed there: your link->6.1 JDBC. There is enough info only about JPA.

**Comment by dstepanov** [comment]
It looks like JOINs section is missing for JDBC/R2DBC. In Micronaut JDBC all queries are native. The criteria usage is included in the section 6.4

**Reply by ArthurHarkivsky** [reply]
1. Yes, missed, thank you! Will it be added soon or much later? +- when?
2. Fixed proposition: CUSTOM native queries with pagination using Micronaut JDBC lib.
3. Thank you! Is it mixing? Does Micr. JDBC lib using Hibernate here? 
```
@JdbcRepository(dialect = Dialect.H2)
public interface PersonRepository extends CrudRepository<Person, Long>, JpaSpecificationExecutor<Person> {
}
```

**Reply by dstepanov** [reply, marked as answer]
1. Please create an issue for it to be added
2. You can create an issue to add that too
3. Micronaut JDBC/R2DBC supports translating JPA annotations into our variations, and we also partially implement JPA criteria API. (It doesn't depend on Hibernate)
